modules/shortcodes/email-subscribe.php

```diff
@@ -0,0 +1,233 @@
+<?php
+/**
+ * WARNING: This file is distributed verbatim in Jetpack.
+ * There should be nothing WordPress.com specific in this file.
+ *
+ * @hide-in-jetpack
+ **/
+
+/**
+ * Class Jetpack_Email_Subscribe
+ * This class encapsulates shortcode for subscribing to a MailChimp list.
+ * It displays a simple signup form that gets an email address from the user and signs him for a list
+ * selected in "Sharing" section in calypso.
+ * Other Email services can be implemented as well in the future.
+ */
+class Jetpack_Email_Subscribe {
+
+	private static $shortcode = 'jetpack-email-subscribe';
+
+	private static $css_classname_prefix = 'jetpack-email-subscribe';
+
+	private static $option_name = 'jetpack_mailchimp';
```

Comment:
This is generally very debateable.

[We plan to add different "email subscription engines"](https://github.com/Automattic/jetpack/pull/10597#discussion_r236598803), so the shortcode could be made flexible, since that is the pattern with shortcodes. We dont want to have too many of them, because people have to remember about them. 

The case with blocks is exactly the opposite - they serve as a discovery method, so we plan to make seperate blocks for each service, powered by the code behind the shortcode.

Reply:
cc: @mapk 

I think we should be careful with introducing too many different blocks here. How is this block different from our current subscription block? Instead of adding another "Mailchimp" subscription block why not add options or setting that turn the current subscription block into the one that is powered by Mailchimp?